core/claims/claimAuthorizeKSignBabyJub_test.go

```diff
@@ -19,28 +20,88 @@ func TestClaimAuthorizeKSignBabyJub(t *testing.T) {
 	}
 	// Add input data to the test vector
 	if generateTest {
-		testgen.SetTestValue("privateKey", "28156abe7fe2fd433dc9df969286b96666489bac508612d0e16593e944c4f69f")
+		testgen.SetTestValue("0_privateKey", "28156abe7fe2fd433dc9df969286b96666489bac508612d0e16593e944c4f69f")
 	}
 	// Create new claim
 	var k babyjub.PrivateKey
-	hexK := testgen.GetTestValue("privateKey").(string)
+	hexK := testgen.GetTestValue("0_privateKey").(string)
 	if _, err := hex.Decode(k[:], []byte(hexK)); err != nil {
 		panic(err)
 	}
 	pk := k.Public()
 	c0 := NewClaimAuthorizeKSignBabyJub(pk)
+	assert.True(t, merkletree.CheckEntryInField(*c0.Entry()))
 	c0.Version = 1
 	e := c0.Entry()
 	// Check claim against test vector
-	checkClaim(e, t)
+	testgen.CheckTestValue(t, "0_HIndex", e.HIndex().Hex())
+	testgen.CheckTestValue(t, "0_HValue", e.HValue().Hex())
+	testgen.CheckTestValue(t, "0_dataString", e.Data.String())
 	dataTestOutput(&e.Data)
 	c1 := NewClaimAuthorizeKSignBabyJubFromEntry(e)
 	c2, err := NewClaimFromEntry(e)
 	assert.Nil(t, err)
 	assert.Equal(t, c0, c1)
 	assert.Equal(t, c0, c2)
+	assert.True(t, merkletree.CheckEntryInField(*e))
 	// Stop test (write new test vector if needed)
 	if err := testgen.StopTest(); err != nil {
 		panic(fmt.Errorf("Error stopping test: %w", err))
 	}
 }
+func TestClaimAuthorizeKSignBabyJub1(t *testing.T) {
+	// If generateTest is true, the checked values will be used to generate a test vector
+	generateTest := false
+	// Init test
+	if err := testgen.InitTest("claimAuthorizeKSignBabyJub", generateTest); err != nil {
+		panic(fmt.Errorf("error initializing test data: %w", err))
+	}
+	// Add input data to the test vector
+	if generateTest {
+		testgen.SetTestValue("1_privateKey", "28156abe7fe2fd433dc9df969286b96666489bac508612d0e16593e944c4f69f")
```

Comment:
this is the same private key as in https://github.com/iden3/go-iden3-core/pull/264/commits/0de37bd015ba1c181e5f3c9fee1007c97c67a821#diff-0473308ee2b6d9453c4aa6d815355514R23

Why are there two tests here that do nearly the same?

Reply:
Right, the idea was that the second was using `9b3260823e7b07dd26ef357ccfed23c10bcef1c85940baa3d02bbf29461bbbbe` as privK. Was changed in the testvector json file, but not in the testvector generator. Updated now

Reply:
If that's the case, could you avoid repeating the 38 lines of the test function that are the same? Thanks!